File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashMap = {};
    
    for([index, element] of nums.entries()){

        let complementaryPair = target - element;
        if(hashMap[complementaryPair] !== undefined)
            return [index, hashMap[complementaryPair]];
        else
        {
            hashMap[element] = index;
        }
    }
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        Dictionary<int, int> numbers = new Dictionary<int, int>();
        
        for(int i = 0; i < nums.Length; i++){
            if(numbers.TryGetValue(target - nums[i], out int val)){
                int[] arr = {i, val};
                return arr;
            }
            numbers.TryAdd(nums[i], i);
        }
        int[] fall = {0,0};
        return fall;
    }
}